torch/csrc/jit/codegen/cuda/test/test_gpu_fused_reduction.cpp

```diff
@@ -2148,6 +2150,309 @@ TEST_F(NVFuserTest, FusionCrossIterationGroupedGridAllreduce4_CUDA) {
+            transform_ref_rf->domain()->domain().end(),
+            [](auto id) {
+              return id->getParallelType() == ParallelType::Vectorize;
+            }));
+    transform_ref_rf->axis(vec_id)->parallelize(ParallelType::Serial);
+
+    int unswitch_id = std::distance(
+        transform_ref_rf->domain()->domain().begin(),
+        std::find_if(
+            transform_ref_rf->domain()->domain().begin(),
+            transform_ref_rf->domain()->domain().end(),
+            [](auto id) {
+              return id->getParallelType() == ParallelType::Unswitch;
+            }));
+    transform_ref_rf->axis(unswitch_id)->parallelize(ParallelType::Serial);
+
+    scheduler_utils::parallelizeAllLike(
+        transform_ref_rf, ir_utils::allTvs(&fusion));
+
+    ParallelType vec_pt = ParallelType::Vectorize;
+    tv1->axis(vec_id)->parallelize(vec_pt);
+    tv9->axis(vec_id)->parallelize(vec_pt);
+
+    transform_ref->axis(vec_id)->parallelize(ParallelType::Group);
+
+    transform_ref_rf->axis(unswitch_id)->parallelize(ParallelType::Unswitch);
+
+    InlinePropagator inline_propagator(
+        transform_ref_rf, -1, ComputeAtMode::MostInlined);
+    MaxRootDomainInfoSpanningTree(transform_ref_rf)
+        .traverse(&inline_propagator);
+
+    // Make sure the reduction expr is converted to GroupedGridReduciton
+    // and the non-reduction domains of the output TV are either
+    // grouped or parallelized
+    GpuLower gpulw(&fusion);
+    bool validated = false;
+    for (auto expr : KernelExprVisitor::getAllExprs(gpulw.kernel())) {
+      auto grouped_grid_reduction =
+          dynamic_cast<kir::GroupedGridWelford*>(expr);
+      validated = true;
+    }
+    TORCH_CHECK(
+        validated, "Invalid lowered kernel. No GroupedGridWelford found.");
+
+    auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+    at::manual_seed(0);
+
+    const std::vector<int64_t> input_shape{
+        params.N, params.H, params.W, params.C};
+    auto t0 = at::randn(input_shape, options);
+
+    FusionExecutor fe;
+    fe.compileFusion(&fusion, {t0});
+    auto cg_outputs = fe.runFusion({t0});
+
+    auto t1 = t0.to(at::kDouble);
+    auto t2 = t1.mean({0, 1, 2}).unsqueeze(0).unsqueeze(0).unsqueeze(0);
+    auto t3 =
+        at::var(t1, {0, 1, 2}, false).unsqueeze(0).unsqueeze(0).unsqueeze(0);
+    auto t4 = params.N * params.H * params.W;
+    auto ref = (t1 - t2 + (t3 * t4));
+
+    testValidate(&fusion, cg_outputs, {t0}, {ref}, __LINE__, __FILE__, "");
+  };
+
+  std::vector<Params> base_params;
+  base_params.push_back({256, 7, 7, 1, 8, 32, 2, 32, 4});
+  base_params.push_back({256, 7, 7, 1, 16, 16, 4, 50, 4});
+  base_params.push_back({128, 7, 7, 1, 16, 16, 4, 32, 4});
+  base_params.push_back({128, 14, 14, 1, 16, 16, 4, 32, 1});
+  base_params.push_back({128, 14, 14, 1, 16, 16, 2, 64, 2});
+  base_params.push_back({128, 14, 14, 1, 8, 32, 4, 50, 4});
+  base_params.push_back({128, 14, 14, 1, 8, 32, 2, 50, 4});
```

Comment:
Are there any GPUs we frequently run tests on that this wouldn't fit on?

Reply:
Forgot to add a check. Thanks for the comment.